src/testSimulator/DataProvider.java

```diff
@@ -0,0 +1,245 @@
+package testSimulator;
+
+import java.util.ArrayList;
+import java.util.InputMismatchException;
+import java.util.Scanner;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class DataProvider {
+
+	private String regex = "-?\\d+x\\^2\\+?-?\\d+x\\+?-?\\d+=0";
+	private int totalNumberOfStudents;
+	private int numberOfBStudents;
+	private int numberOfCStudents;
+	private int numberOfDStudents;
+	private int numberOfEStudents;
+	private int numberOfFStudents;
+	private int numberOfClasses;
+	private String equasion;
+
+	public int getTotalNumberOfStudents() {
+		return totalNumberOfStudents;
+	}
+
+	public void setTotalNumberOfStudents(int totalNumberOfStudents) {
+		this.totalNumberOfStudents = totalNumberOfStudents;
+	}
+
+	public int getNumberOfBStudents() {
+		return numberOfBStudents;
+	}
+
+	public void setNumberOfBStudents(int numberOfBStudents) {
+		this.numberOfBStudents = numberOfBStudents;
+	}
+
+	public int getNumberOfCStudents() {
+		return numberOfCStudents;
+	}
+
+	public void setNumberOfCStudents(int numberOfCStudents) {
+		this.numberOfCStudents = numberOfCStudents;
+	}
+
+	public int getNumberOfDStudents() {
+		return numberOfDStudents;
+	}
+
+	public void setNumberOfDStudents(int numberOfDStudents) {
+		this.numberOfDStudents = numberOfDStudents;
+	}
+
+	public int getNumberOfEStudents() {
+		return numberOfEStudents;
+	}
+
+	public void setNumberOfEStudents(int numberOfEStudents) {
+		this.numberOfEStudents = numberOfEStudents;
+	}
+
+	public int getNumberOfFStudents() {
+		return numberOfFStudents;
+	}
+
+	public void setNumberOfFStudents(int numberOfFStudents) {
+		this.numberOfFStudents = numberOfFStudents;
+	}
+
+	public String getEquasion() {
+		return equasion;
+	}
+
+	public void setEquasion(String equasion) {
+		this.equasion = equasion;
+	}
+
+	public int getNumberOfClasses() {
+		return numberOfClasses;
+	}
+
+	public void getClassroomInformationFromUser(School school) {
+		ArrayList<Classroom> classrooms = new ArrayList<Classroom>();
+		askNumOfClasses();
+		for (int i = 0; i < this.numberOfClasses; i++) {
+			System.out.println("Podaj dane dla " + (i + 1) + " klasy");
+			do {
+				askNumOfAllStudents();
+				askNumOfBStudents();
+				askNumOfCStudents();
+				askNumOfDStudents();
+				askNumOfEStudents();
+				askNumOfFStudents();
+			} while (!this.isDataValid());
+
+			do {
+				String equasionToCheck = askEquasion();
+				this.setEquasion(equasionToCheck);
+			} while (!this.checkEquasion(regex, equasion));
```

Comment:
As an argument you're passing a value that is a field in this class. Either remove from class level, define locally and pass as an argument to that method or don't remove it from the class level, do not pass it as an argument and simply refer to that field from the method itself.